validation/src/shared_table/mod.rs

```diff
@@ -287,16 +287,17 @@ impl<Fetch: Future> ParachainWork<Fetch> {
 	pub fn prime<P: ProvideRuntimeApi>(self, api: Arc<P>)
 		-> PrimedParachainWork<
 			Fetch,
-			impl Send + FnMut(&BlockId, &Collation) -> Result<OutgoingMessages, ()>,
+			impl Send + FnMut(&BlockId, &Collation, &CandidateReceipt) -> Result<(OutgoingMessages, ErasureChunk), ()>,
 		>
 		where
 			P: Send + Sync + 'static,
 			P::Api: ParachainHost<Block>,
 	{
 		let max_block_data_size = self.max_block_data_size;
 		let relay_parent = self.relay_parent.clone();
+		let local_index = self.local_index;
 
-		let validate = move |id: &_, collation: &_| {
+		let validate = move |id: &_, collation: &_, receipt: &_| {
```

Comment:
i find it really strange that this requires the `collation`.

this function is in the context of "i got a receipt and PoVBlock from another validator".
so we're going from `receipt -> (collation, receipt)` and then doing validation of both?

there should be a utility method that just validates `(Receipt, PoVBlock)`.